### PR TITLE
Add a config.HOSTNAME var to avoid host header attacks

### PR DESCRIPTION
**Pull request by estellecomment**
Add a HOSTNAME_WITH_PROTOCOL config var, that will be used to send and display links, instead of using the host header from the request.
This avoids host header poisoning attacks (detected by CodeQL : https://github.com/betagouv/audioconf/security/code-scanning/11?query=ref%3Arefs%2Fheads%2Fmain)

By default, HOSTNAME_WITH_PROTOCOL is http://localhost:${PORT}

We use HOSTNAME_WITH_PROTOCOL (instead of HOSTNAME and PROTOCOL) separately, because that's the var that scalingo review apps have. So we use it for all scalingo apps for simplicity.

This PR adds a scalingo.json file (with which we configure the HOSTNAME_WITH_PROTOCOL vars). When this file is present, the app.json file is ignored, so this PR removes it.

 - [x] add HOSTNAME_WITH_PROTOCOL var to prod instance
 - [x] add HOSTNAME_WITH_PROTOCOL var to staging instance
 - [x] how to get hostname in review apps ?
 - [x] update docker files to avoid breaking docker installs

### CODE REVIEW

**Review by eml-trm — APPROVED** [review]
good to me ;)

**Review by polomarcus — APPROVED** [review]

**Comment by jdauphant** [comment]
Comment sont géré les pull request ?

Pour info, un reverse proxy empéche l'attaque de ce genre (c'est le cas sur Scalingo)

File: scalingo.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "AudioConf",
+  "env": {
+    "HOSTNAME_WITH_PROTOCOL": {
+      "description": "Url with protocol, ex: https://my-scalingo-app-pr36.scalingo.io",
+      "generator": "url"
```
Comment (jdauphant) [inline scalingo.json:6]: ça génére l'url de scalingo pour la review app ?

File: .env.sample [hunk 2]
```diff
@@ -34,7 +34,8 @@ OVH_ROOM_PHONE_NUMBER=CHANGE_ME
 
 # HTTP
 #PORT=8080
-PROTOCOL="http"
+# Hostname : the links sent in email will use this hostname. Default is http://localhost:$PORT
+HOSTNAME_WITH_PROTOCOL="https://myapp.com"
```
Comment (jdauphant) [inline .env.sample:38]:
```suggestion
#HOSTNAME_WITH_PROTOCOL="https://myapp.com"
```
ou sinon mettre une valeur qui marche en local

File: scalingo.json (hunk 1 above)
Comment (estellecomment) [reply scalingo.json:6]: oui

**Pull request merged**